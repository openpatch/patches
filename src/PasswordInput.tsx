import { Input, InputProps } from "./Input";

export type PasswordInputProps = {} & InputProps;

export const PasswordInput = ({ ...props }: PasswordInputProps) => {
  return <Input type="password" {...props} />;
};
