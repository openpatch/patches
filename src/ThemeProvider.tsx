import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
