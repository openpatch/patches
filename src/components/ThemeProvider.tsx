import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
