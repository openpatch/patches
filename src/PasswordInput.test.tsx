
import { render } from "@testing-library/react";
import { PasswordInput, PasswordInputProps } from "./PasswordInput";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: PasswordInputProps) =>
render(
    <ThemeProvider>
        <PasswordInput {...props} />
    </ThemeProvider>
);

test("should render", () => {
   renderWithTheme();
});

test("should match snapshot", () => {
    const { container } = renderWithTheme();
    expect(container).toMatchSnapshot();
});