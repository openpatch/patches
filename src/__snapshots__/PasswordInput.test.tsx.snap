// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  border-radius: 4px;
  padding: 8px 4px;
  border-color: #E4E7EB;
  border-style: solid;
  border-width: 1px;
  box-sizing: border-box;
  width: 100%;
}

.emotion-0::-webkit-input-placeholder {
  color: #7B8794;
}

.emotion-0::-moz-placeholder {
  color: #7B8794;
}

.emotion-0:-ms-input-placeholder {
  color: #7B8794;
}

.emotion-0::placeholder {
  color: #7B8794;
}

.emotion-0:focus {
  outline-color: transparent;
  outline-width: 0;
  border-color: #0C6B58;
  box-shadow: 0 0 0 3px rgba(66, 153, 225, 0.5);
  z-index: 1;
}

.emotion-0:disabled {
  background-color: #F5F7FA;
}

<div>
  <div
    id="patches-root"
  >
    <input
      aria-invalid="false"
      class="emotion-0 emotion-1"
      type="password"
      value=""
    />
  </div>
</div>
`;
